gammapy/irf/psf/map.py

```diff
@@ -243,14 +262,19 @@ def get_psf_kernel(
             across all energies is used. The radius can be overwritten using
             the `max_radius` argument. Default is 0.999.
         factor : int, optional
-            Oversampling factor to compute the PSF. Default is 4.
-
+            Oversampling factor to compute the PSF.
+            If None it is given by PSF_UPSAMPLING_FACTOR which is 4 by default.
+            If PSF_UPSAMPLING_FACTOR is set to None it will be automatically estimated.
         Returns
         -------
         kernel : `~gammapy.irf.PSFKernel`
             The resulting kernel.
         """
-        # TODO: try to simplify...is the oversampling needed?
+        if factor is None:
```

Comment:
I think this behavior is confusing. What about:

- Introduce a new `precision_factor` and `factor_max` argument
- Forward both to `_psf_upsampling_factor`
- Deprecate `factor`...